### DISCUSSION TITLE
Resource name prefix auto added in CodePipeline

### DISCUSSION THREAD

**Discussion by byongwu** [aws/aws-cdk#22899] [Q&A]
Hi,

### TL;DR
- I would like to disable the automatic resource name prefix when CDK + CodePipeline is used.

### Background
- I am working on customer's project with CDK (TypeScript) + CodePipeline.
- One small issue is that, when I am working with CodePipeline, I get automatic resource name prefix.
- For example, this is the resource hierarchy.
- CUSTOMER-PREFIX-GR: stack group (GR)
  - CUSTOMER-PREFIX-GR-ST: a stack (ST) in a group
     - CUSTOMER-PREFIX-GR-STSecretMgrAccessRoleABCD1234-ABCD1234: IAM role created

### Question
- However, what our customer want is this
  - CUSTOMER-PREFIX-SecretMgrAccessRoleABCD1234-ABCD1234
- Please note that `GR-ST` part of the prefix is taken out.
- How can I achieve this goal?
- I would like to disable the automatic resource name prefix

### Appendix. Code Used
- This is the code used for stack group (GR)

```typescript
export class MainPipelineStage extends cdk.Stage {
  constructor(scope: Construct, id: string, props?: cdk.StageProps) {
    super(scope, id, props);
    const stackGroupDMS = new StackGroupDMS(this, "GR");
  }
}

export class StackGroupDMS extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);
    new StackDMS(this, "ST");
  }
}

export class StackDMS extends Construct {
  constructor(scope: Construct, id: string  ) {
    super(scope, id);
    // ... provisioning
  }
}
```

Thanks in advance!

**Comment by byongwu** [comment]
I am not an expert of CDK internal. But I did some code search.

It looks to me that [uniqueId()](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/pipelines/lib/helpers-internal/graph.ts#L28) in pipelines is the relevant code making the chained resource name in CodePipeline.

```typescript
  /**
   * A graph-wide unique identifier for this node. Rendered by joining the IDs
   * of all ancestors with hyphens.
   */
  public get uniqueId(): string {
    return this.ancestorPath(this.root).map(x => x.id).join('-');
  }
```

Would it be possible to add a control flag to disable joining all the ancestorPath?

**Comment by philmod-h** [comment]
In my case, it doesn't look like all the resources have the pipeline ID added to their ID?

Did you find a solution to that problem?
